WebGL: failed to import and connect to temple

1. When trying to import 1.3.1:
```
Library\PackageCache\com.trilitech.tezos-unity-sdk@a5b7a15f52\Runtime\Scripts\BeaconSDK\BeaconConnectorWebGl.cs(12,38): error CS0535: 'BeaconConnectorWebGl' does not implement interface member 'IBeaconConnector.RequestTezosSignPayload(SignPayloadType, string)'
```

Workaround: change enum type to int

2. When trying to connect to Temple: stays as operation loading and does not prompt for wallet connection.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 
+## [unreleased]
+### Fixed
+- [Bug](https://github.com/trilitech/tezos-unity-sdk/issues/57) with BeaconConnectorWebGl
+
+
 ## [1.3.1] - 2023-04-27
 ### Changed
 - Beacon SDK version to [1.0.23](https://github.com/baking-bad/beacon-dotnet-sdk/releases/tag/v1.0.23)

--- a/Runtime/Scripts/BeaconSDK/BeaconConnectorWebGl.cs
+++ b/Runtime/Scripts/BeaconSDK/BeaconConnectorWebGl.cs
@@ -1,121 +1,123 @@
 #if UNITY_WEBGL
 using System;
 using System.Runtime.InteropServices;
+using Beacon.Sdk.Beacon.Sign;
 using UnityEngine;
 
 namespace BeaconSDK
 {
-	/// <summary>
-	/// WebGL implementation of the BeaconConnector.
-	/// Binds the functions implemented inside the file BeaconConnection.jslib
-	/// </summary>
-	public class BeaconConnectorWebGl : IBeaconConnector
-	{
-		#region Bridge to external functions
+    /// <summary>
+    /// WebGL implementation of the BeaconConnector.
+    /// Binds the functions implemented inside the file BeaconConnection.jslib
+    /// </summary>
+    public class BeaconConnectorWebGl : IBeaconConnector
+    {
+        #region Bridge to external functions
 
-		[DllImport("__Internal")]
-		private static extern void JsSetNetwork(string network, string rpc);
+        [DllImport("__Internal")]
+        private static extern void JsSetNetwork(string network, string rpc);
 
-		[DllImport("__Internal")]
-		private static extern void JsConnectAccount();
+        [DllImport("__Internal")]
+        private static extern void JsConnectAccount();
 
-		[DllImport("__Internal")]
-		private static extern void JsSwitchAccounts();
+        [DllImport("__Internal")]
+        private static extern void JsSwitchAccounts();
 
-		[DllImport("__Internal")]
-		private static extern void JsRemovePeer();
+        [DllImport("__Internal")]
+        private static extern void JsRemovePeer();
 
-		[DllImport("__Internal")]
-		private static extern void JsSendMutezAsString(string amount, string address);
+        [DllImport("__Internal")]
+        private static extern void JsSendMutezAsString(string amount, string address);
 
-		[DllImport("__Internal")]
-		private static extern void JsSendContractCall(string destination, string amount, string entryPoint, string arg);
+        [DllImport("__Internal")]
+        private static extern void JsSendContractCall(string destination, string amount, string entryPoint, string arg);
 
-		[DllImport("__Internal")]
-		private static extern void JsReset();
-		
-		[DllImport("__Internal")]
-		private static extern void JsSignPayload(int signingType, string payload);
-		
-		[DllImport("__Internal")]
-		private static extern string JsGetActiveAccountAddress();
-		
-		#endregion
-		
-		private string _activeAccountAddress;
+        [DllImport("__Internal")]
+        private static extern void JsReset();
 
-		public void SetNetwork(string network, string rpc)
-		{
-			JsSetNetwork(network, rpc);
-		}
+        [DllImport("__Internal")]
+        private static extern void JsSignPayload(int signingType, string payload);
 
-		public void ConnectAccount()
-		{
-			JsConnectAccount();
-		}
+        [DllImport("__Internal")]
+        private static extern string JsGetActiveAccountAddress();
 
-		public void CallContract(string destination, string entryPoint, string arg, long amount = 0)
-		{
-			JsSendContractCall(destination, amount.ToString(), entryPoint, arg);
-		}
+        #endregion
 
-		public void DisconnectAccount()
-		{
-			JsRemovePeer();
-		}
+        private string _activeAccountAddress;
+
+        public void SetNetwork(string network, string rpc)
+        {
+            JsSetNetwork(network, rpc);
+        }
+
+        public void ConnectAccount()
+        {
+            JsConnectAccount();
+        }
+
+        public void CallContract(string destination, string entryPoint, string arg, long amount = 0)
+        {
+            JsSendContractCall(destination, amount.ToString(), entryPoint, arg);
+        }
+
+        public void DisconnectAccount()
+        {
+            JsRemovePeer();
+        }
+
 /*
 		public void SendMutez(long amount, string address)
 		{
 			JsSendMutezAsString(amount.ToString(), address);
 		}
 */
-		public void SwitchAccounts()
-		{
-			JsSwitchAccounts();
-		}
+        public void SwitchAccounts()
+        {
+            JsSwitchAccounts();
+        }
 
-		public void Reset()
-		{
-			JsReset();
-		}
+        public void Reset()
+        {
+            JsReset();
+        }
 
-		public string GetActiveAccountAddress()
-		{
-			return JsGetActiveAccountAddress();
-		}
+        public string GetActiveAccountAddress()
+        {
+            return JsGetActiveAccountAddress();
+        }
 
-		public void RequestHandshake()
-		{
-			throw new NotImplementedException();
-		}
+        public void RequestHandshake()
+        {
+            throw new NotImplementedException();
+        }
 
-		public void SetActiveAccountAddress(string address)
-		{
-			_activeAccountAddress = address;
-		}
+        public void SetActiveAccountAddress(string address)
+        {
+            _activeAccountAddress = address;
+        }
 
-		public void RequestTezosPermission(string networkName = "", string networkRPC = "")
-		{
-			Debug.Log("ConnectAccount executes RequestPermissions");
-			throw new NotImplementedException();
-		}
+        public void RequestTezosPermission(string networkName = "", string networkRPC = "")
+        {
+            Debug.Log("ConnectAccount executes RequestPermissions");
+            throw new NotImplementedException();
+        }
 
-		public void RequestTezosOperation(string destination, string entryPoint = "default", string arg = null,
-			ulong amount = 0, string networkName = "", string networkRPC = "")
-		{
-			JsSetNetwork(networkName, networkRPC);
-			JsSendContractCall(destination, amount.ToString(), entryPoint, arg);
-		}
+        public void RequestTezosOperation(string destination, string entryPoint = "default", string arg = null,
+            ulong amount = 0, string networkName = "", string networkRPC = "")
+        {
+            JsSetNetwork(networkName, networkRPC);
+            JsSendContractCall(destination, amount.ToString(), entryPoint, arg);
+        }
 
-		public void RequestTezosSignPayload(int signingType, string payload)
-		{
-			JsSignPayload(signingType, payload);
-		}
+        public void RequestTezosSignPayload(SignPayloadType signingType, string payload)
+        {
+            JsSignPayload((int)signingType, payload);
+        }
 
-		public void RequestTezosBroadcast(string signedTransaction, string networkName = "", string networkRPC = "")
-		{
-			throw new NotImplementedException();
-		}
-	}
+        public void RequestTezosBroadcast(string signedTransaction, string networkName = "", string networkRPC = "")
+        {
+            throw new NotImplementedException();
+        }
+    }
 }
 #endif

--- a/Runtime/Scripts/BeaconSDK/BeaconMessageReceiver.cs
+++ b/Runtime/Scripts/BeaconSDK/BeaconMessageReceiver.cs
@@ -74,7 +74,7 @@ namespace BeaconSDK
 		public IEnumerator ContractCallInjection(string uri, string transactionHash)
 		{
 			var success = false;
-			var timeout = 30f; // seconds
+			const float timeout = 30f; // seconds
 			var timestamp = Time.time;
 			Debug.Log("Operation injected into blockchain");
 			

